drivers/storage/ide/uniata/id_probe.cpp

```diff
@@ -1942,6 +1942,7 @@ UniataClaimLegacyPCIIDE(
 {
     NTSTATUS status;
     PCM_RESOURCE_LIST resourceList = NULL;
+    PCM_RESOURCE_LIST oldResList   = NULL;
```

Comment:
Ah apologies this is my first time contributing, I do have some questions though

1. This fix is not specific to ReactOS, The memory leaks would be present even if this code were to run on Windows NT or ReactOS. Do we add #ifdef to 3rd party drivers for every change? If it was something related specific to ReactOS I would understand but this is just a dangling pointer which is freed.

2.  Unless we push these changes upstream (to UniATA) this part of the code is now part of ReactOS yes? Why do we add #ifdef(s)

Reply:
The best thing to do would be upstream integration ofc, especially if the drivers failed similarly on Windows. We would then not need the ifdefs on our side.
But often upstream integration is taking considerable amount of time, or isn't possible at all (uniata maintainer named "Altar" seems to be not that active anymore).
In such cases we do use the #ifdef guards to highlight our diff to the upstream repo.
Hoping for them to be a temporary thing. But more often than we like, they are getting a permanent thing.
That is still better than not adding them.

Reply:
In any case, Alter is alive because http://alter.org.ua/ru/  shows updates of some of his stuff on this year (2023).

Reply:
Cc @Alter-1 could you check this UniATA bugfix? :)

Reply:
> In any case, Alter is alive because http://alter.org.ua/ru/ shows updates of some of his stuff on this year (2023).
> Cc @Alter-1 could you check this UniATA bugfix? :)

I wonder if this is not on his priority list since there is war and electricity is not stable in his state.

Reply:
Done! The code is now under `#ifdef __REACTOS__` block

Reply:
Thanks. We will wait one or two weeks, hoping for @Alter-1 to show up. Then we will merge it most likely.
Question: What are the **exact reproduction steps** to make it leak, and how did you confirm it to stop leaking then?

Reply:
> Question: What are the exact reproduction steps to make it leak, and how did you confirm it to stop leaking then?

To reproduce:  Create VM with storage attached to PCI IDE controller (PXII3/4 under VBox). Memory should leak while UniATA is loaded and enumerates on the PCI bus

To confirm leak is plugged: Ran under WinDBG, observing the dangling pointer and the new pointer allocated by HalAssignSlotResources, ensure both pointers are freed